src/optimizer/cost_and_stats_calculator.cpp

```diff
@@ -43,18 +50,90 @@ std::shared_ptr<TableStats> generateOutputStat(
 
       // TODO: Deal with or add column stats for complex expressions like a+b
       if (expr->GetExpressionType() == ExpressionType::VALUE_TUPLE) {
-        auto column_name =
-            (std::dynamic_pointer_cast<expression::TupleValueExpression>(expr))
-                ->GetColumnName();
+        auto tv_expr =
+            std::dynamic_pointer_cast<expression::TupleValueExpression>(expr);
+        auto column_name = tv_expr->GetColumnName();
+        if (data_table == nullptr) {
+          column_name = tv_expr->GetTableName() + "." + column_name;
+        }
         auto column_stat = input_table_stats->GetColumnStats(column_name);
         if (column_stat != nullptr) {
+          if (data_table != nullptr) {
+            // add table_name for base table
+            column_stat->column_name =
+                data_table->GetName() + "." + column_stat->column_name;
+          }
           output_column_stats.push_back(column_stat);
         };
       }
     }
   }
   auto output_table_stats = std::make_shared<TableStats>(
       input_table_stats->num_rows, output_column_stats);
+
+  // Set sampler
+  output_table_stats->SetTupleSampler(input_table_stats->GetSampler());
+
+  // Add indexes to index map for base table
+  if (data_table != nullptr) {
+    for (oid_t i = 0; i < data_table->GetIndexCount(); i++) {
+      auto index = data_table->GetIndex(i);
+      // If the index is for multiple columns, we don't add
+      if (index->GetMetadata()->GetKeyAttrs().size() > 1) {
+        continue;
+      }
+      auto column_id = index->GetMetadata()->GetKeyAttrs()[0];
+      // If the index column does not exist in the property column, we don't add
+      if (input_table_stats->GetColumnStats(column_id)->column_name.find(".") ==
```

Comment:
Same. Would be good to write a separate function for this.

Reply:
Agree with you. I think another solution is to change the way we store the column name in the ColumnStats. Therefore, we don't need to to the transition every time. Which one do you prefer for now?